superset/views.py

```diff
@@ -1465,6 +1469,58 @@ def explore(self, datasource_type, datasource_id):
                 userid=g.user.get_id() if g.user else ''
             )
 
+    @api
+    @has_access_api
+    @expose("/filter/<datasource_type>/<datasource_id>/<column>/")
+    def filter(self, datasource_type, datasource_id, column):
+        """
+        Endpoint to retrieve values for specified column.
+        :param datasource_type: Type of datasource e.g. table
+        :param datasource_id: Datasource id
+        :param column: Column name to retrieve values for
+        :return:
+        """
+        # TODO: Cache endpoint by user, datasource and column
+        error_redirect = '/slicemodelview/list/'
+        datasource_class = models.SqlaTable \
+            if datasource_type == "table" else models.DruidDatasource
+
+        datasource = db.session.query(
+            datasource_class).filter_by(id=datasource_id).first()
+
+        if not datasource:
+            flash(__("The datasource seems to have been deleted"), "alert")
```

Comment:
we have a constant, `DATASOURCE_MISSING_ERR`, for this error message here:
https://github.com/airbnb/superset/blob/master/superset/views.py#L81